pkg/controller/metadata/metadata_controller.go

```diff
@@ -385,8 +406,24 @@ func (r *ReconcileMetadata) Reconcile(request reconcile.Request) (reconcile.Resu
 		}
 	} else if err != nil {
 		return reconcile.Result{}, err
-	} else {
-		// TODO: Update resource if changed
+	} else if foundService.ObjectMeta.Annotations[versionAnnotation] != currentVersion {
+		log.Info("Updating Service", "namespace", metadataService.Namespace, "name", metadataService.Name)
+
+		foundService.Spec.Selector = metadataLabels
+		foundService.Spec.Ports = []corev1.ServicePort{
+			{
+				Protocol:   "TCP",
+				Port:       80,
+				Name:       "http",
+				TargetPort: intstr.FromInt(80),
+			},
+		}
+
+		err = r.Update(context.TODO(), foundService)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
```

Comment:
Could an `info` log message be added to indicate successful completion?

Reply:
yep